deepspeech.go

```diff
@@ -63,9 +63,51 @@ type sliceHeader struct {
 // buffer     A 16-bit, mono raw audio signal at the appropriate sample rate.
 // bufferSize The number of samples in the audio signal.
 // sampleRate The sample-rate of the audio signal.
-// TODO Make sure the C string is cleaned properly
 func (m *Model) SpeechToText(buffer []int16, bufferSize, sampleRate uint) string {
-	return C.GoString(C.STT(m.w, (*C.short)(unsafe.Pointer((*sliceHeader)(unsafe.Pointer(&buffer)).Data)), C.uint(bufferSize), C.uint(sampleRate)))
+	str := C.STT(m.w, (*C.short)(unsafe.Pointer((*sliceHeader)(unsafe.Pointer(&buffer)).Data)), C.uint(bufferSize), C.uint(sampleRate));
+	retval := C.GoString(str)
+	C.FreeString(str)
```

Comment:
I would move this one line above using a `defer` keyword (same goes for other use of `FreeString`):

```go
	str := C.STT(m.w, (*C.short)(unsafe.Pointer((*sliceHeader)(unsafe.Pointer(&buffer)).Data)), C.uint(bufferSize), C.uint(sampleRate))
	defer C.FreeString(str)
	retval := C.GoString(str)
```

Reply:
Okay, but why ? It is my understanding that `C.GoString()` will not rely anymore on the string, what's the use of `defer` ? Docs says it's evaluated in place but ran at the end of the function.

Reply:
`defer` is used here to make sure this action will be executed whatever there's after. `FreeString` will still be executed after `GoString` but that way we know right after creating `str` that it will be freed properly.

Reply:
Makes sense